src/libraries/icubmod/embObjMotionControl/eomcParser.cpp

```diff
@@ -1272,7 +1337,39 @@ bool Parser::getCorrectPidForEachJoint(PidInfo *ppids/*, PidInfo *vpids*/, TrqPi
             if(tpids[firstjoint].fbk_PidUnits != tpids[i].fbk_PidUnits ||
                tpids[firstjoint].out_PidUnits != tpids[i].out_PidUnits)
             {
-                yError() << "embObjMC BOARD " << _boardname << "all joints with torque enabled should have same controlunits type. Joint " << firstjoint << " differs from joint " << i;
+                yError() << "embObjMC BOARD " << _boardname << "all joints with TORQUE enabled should have same controlunits type. Joint " << firstjoint << " differs from joint " << i;
+                return false;
+            }
+        }
+    }
+
+    //Here i would check that all joints have same type units in order to create position helper with correct factor.
+
+    //get first joint with enabled position
+    firstjoint = -1;
+    for(int i=0; i<_njoints; i++)
+    {
+        if(ppids[i].enabled)
+        {
+            firstjoint = i;
+            break;
+        }
+    }
+
+    if(firstjoint==-1)
+    {
+        // no joint has position enabed
+        return true;
+    }
+
+    for(int i=firstjoint+1; i<_njoints; i++)
+    {
+        if(ppids[i].enabled)
+        {
+            if(ppids[firstjoint].fbk_PidUnits != ppids[i].fbk_PidUnits ||
+               ppids[firstjoint].out_PidUnits != ppids[i].out_PidUnits)
+            {
+                yError() << "embObjMC BOARD " << _boardname << "all joints with POSITION enabled should have same controlunits type. Joint " << firstjoint << " differs from joint " << i;
```

Comment:
finally, added the error message for position control (not just torque). Also, made the origin of the messages more evident by putting in caps (e.g.: `POSITION`) the type of control that is reporting the error